snap/snapenv/snapenv.go

```diff
@@ -63,6 +63,7 @@ func GetBasicSnapEnvVars(desc interface{}) []string {
 	return fillSnapEnvVars(desc, []string{
 		"SNAP={{.SnapPath}}",
 		"SNAP_DATA=/var{{.SnapPath}}",
+		"SNAP_SHARED_DATA=/var{{.SnapBasePath}}/shared",
```

Comment:
Why are we building this variable here when we have a method in info that returns the exact correct path? We should have the info here and use its methods (templates can do that) instead of inventing a new way to get the same data for each one of these variables.